### PR TITLE
TIG-1895 Oplog tailer

### PR DESCRIPTION
**Pull request by luke-pearson**
Currently this code contains a bit of a hack where the TODO is. Waiting on a new feature from Ryan before merging, however I would like to begin reviewing the overall actor.

### CODE REVIEW

File: src/cast_core/src/RollingCollections.cpp [hunk 1]
```diff
@@ -260,6 +253,65 @@ struct Manage : public RunOperation {
...
+            for (auto&& doc : _cursor.value()) {
+                // Do nothing.
+            }
+            _firstLoop = false;
+        }
+        for (auto&& doc : _cursor.value()) {
+            if (doc["op"].get_utf8().value.to_string() == "c") {
+                auto object = doc["o"].get_document().value;
+                auto it = object.find("create");
+                if (it != object.end()) {
```
Comment (rtimmons) [inline src/cast_core/src/RollingCollections.cpp]: consider negating this and `continue`ing to save some indentation

```c++
if (it == end()) { continue; }
```

same also applies to above

```c++
if (doc["op"].get_utf8().value.to_string() != "c") { continue; }
```

File: src/cast_core/src/RollingCollections.cpp [hunk 2]
```diff
@@ -260,6 +253,65 @@ struct Manage : public RunOperation {
...
+        if (phaseContext.actor()["Threads"].to<int>() != 1) {
+            BOOST_THROW_EXCEPTION(
+                InvalidConfigurationException("OplogTailer can only be run with one thread"));
+        }
+    }
+
+    void run() override {
+        if (_firstLoop) {
+            mongocxx::options::find opts{};
+            opts.cursor_type(mongocxx::cursor::type::k_tailable);
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply src/cast_core/src/RollingCollections.cpp]: I get a fairly weird compile warning when I do that so I'll skip fixing this for now.

File: src/cast_core/src/RollingCollections.cpp (hunk 1 above)
Comment (luke-pearson) [reply src/cast_core/src/RollingCollections.cpp]: I dunno I kinda like it as is.

**Comment by rtimmons** [comment]
This is blocked by [TIG-1984](https://jira.mongodb.org/browse/TIG-1984) which will refine the metrics API for adding synthetic data-points.

**Comment by rtimmons** [comment]
@luke-pearson I've updated the RollingCollection.cpp code here to use the new synthetic-metrics API. (I realize the API is still a bit unpleasant but before optimizing to make this Actor's life easier I want to see how other Actors end up using it.) There is a TODO in there for you. Let me know if any questions or if/when ready for further review.

File: src/cast_core/src/RollingCollections.cpp [hunk 3]
```diff
@@ -50,25 +52,18 @@ struct RunOperation {
     RollingCollectionNames& rollingCollectionNames;
 };
 
+static long getMillisecondsSinceEpoch() {
+    return std::chrono::duration_cast<std::chrono::milliseconds>(
+               std::chrono::system_clock::now().time_since_epoch())
+        .count();
```
Comment (rtimmons) [inline src/cast_core/src/RollingCollections.cpp]: this answers my question in the TODO I added - need to change this to `metrics::clock::now()`

**Comment by luke-pearson** [comment]
@rtimmons I've fixed up the todo and clang formatted everything. Could you please do a final round of review? It'll be good to get this closed.

**Pull request merged**